DFC.ServiceTaxonomy.ApiFunction/Function/Execute.cs

```diff
@@ -0,0 +1,176 @@
+using DFC.ServiceTaxonomy.ApiFunction.Helpers;
+using DFC.ServiceTaxonomy.ApiFunction.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Neo4j.Driver.V1;
+using ILogger = Microsoft.Extensions.Logging.ILogger;
+
+namespace DFC.ServiceTaxonomy.ApiFunction.Function
+{
+    public class Execute
+    {
+        private readonly IOptions<ServiceTaxonomyApiSettings> _serviceTaxonomyApiSettings;
+        private readonly IHttpRequestHelper _httpRequestHelper;
+        private readonly IJsonHelper _jsonHelper;
+        private readonly INeo4JHelper _neo4JHelper;
+        private readonly IFileHelper _fileHelper;
+
+        public Execute(IOptions<ServiceTaxonomyApiSettings> serviceTaxonomyApiSettings, IHttpRequestHelper httpRequestHelper, IJsonHelper jsonHelper, INeo4JHelper neo4JHelper, IFileHelper fileHelper)
+        {
+            _serviceTaxonomyApiSettings = serviceTaxonomyApiSettings ?? 
+                                          throw new ArgumentNullException(nameof(serviceTaxonomyApiSettings));
+
+            _httpRequestHelper = httpRequestHelper ??
+                                 throw new ArgumentNullException(nameof(httpRequestHelper));
+
+            _jsonHelper = jsonHelper ??
+                  throw new ArgumentNullException(nameof(jsonHelper));
+
+            _neo4JHelper = neo4JHelper ?? 
+                           throw new ArgumentNullException(nameof(neo4JHelper));
+
+            _fileHelper = fileHelper ??
+                          throw new ArgumentNullException(nameof(fileHelper));
+        }
+
+        [FunctionName("Execute")]
+        public async Task<IActionResult> Run(
```

Comment:
Ideally we should perform some of these operations concurrently, which would provide a performance boost. Ideally this should be done after breaking the method down (see comment above) for code readability.
At the moment, the sequence is as follows:

`GetRequestBody -> GetCypherQuery -> ExecuteCypherQuery -> GetRecords -> GetResultSummary`

It could be:
```
GetRequestBody \                      / GetRecords
                > ExecuteCypherQuery >
GetCypherQuery /                      \ GetResultSummary
```